impress-org/givewp
# Milestone block now supports a title and description

_**Note: This PR is depended upon work in #5224 - it cannot be merged until that PR is first approved and merged!**_

Resolves #5225

## Description
This PR introduces the title and description fields associated with the Milestone block. Previously, the Milestone block only displayed a customize-able title, this PR expands the Inspector functionality and adds the new `description` attribute so that admin can also modify the description.

_In my own testing, there is a somewhat jarring flicker which occurs between server side renders of the Milestone block in the block editor. While this is annoying, it's not an issue produced specifically by the introduction of the title/description fields, so I am opening a separate issue aimed at addressing that problem._

## Affects
This PR affects the Milestone block registration (adding the `description` attribute), and block editor rendering logic associated specifically with the Milestone block Inspector controls. 

## Visuals
![TitleDescription](https://user-images.githubusercontent.com/5186078/92014616-63e7a900-ed04-11ea-9559-a03ae546fb5f.gif)

<img width="629" alt="Screen Shot 2020-09-02 at 10 03 56 AM" src="https://user-images.githubusercontent.com/5186078/92014588-57fbe700-ed04-11ea-9bf6-b40ad09a7a32.png">

## Pre-review Checklist

## Testing Instructions
1. Create a new page
2. Add the Milestone block
3. Change the title or description
4. See the changes reflected in realtime within the editor
5. Save, and see them also reflected on the frontend
